cmake/kokkos_enable_options.cmake

```diff
@@ -29,6 +29,18 @@ KOKKOS_DEPRECATED_LIST(OPTIONS ENABLE)
 KOKKOS_ENABLE_OPTION(CUDA_RELOCATABLE_DEVICE_CODE  OFF "Whether to enable relocatable device code (RDC) for CUDA")
 KOKKOS_ENABLE_OPTION(CUDA_UVM             OFF "Whether to use unified memory (UM) for CUDA by default")
 KOKKOS_ENABLE_OPTION(CUDA_LDG_INTRINSIC   OFF "Whether to use CUDA LDG intrinsics")
+if(Kokkos_ENABLE_CUDA)
+  # In contrast to other CUDA dependent options CUDA_LAMBDA is ON by default.
+  # It is problematic when CUDA is not enabled, because not only it yields a
+  # bogus warning, it also exports the Kokkos_ENABLE_CUDA_LAMBDA variable and
+  # set it to ON. This if-clause is a crouch that delays our refactoring the
+  # way we declare all options until after we get rid of TriBITS.
```

Comment:
```suggestion
  # In contrast to other CUDA-dependent, options CUDA_LAMBDA is ON by default.
  # That is problematic when CUDA is not enabled because this not only yields a
  # bogus warning, but also exports the Kokkos_ENABLE_CUDA_LAMBDA variable and
  # sets it to ON. This if-clause is a crutch that delays the refactoring of the
  # way we declare all options until after we get rid of TriBITS.
```